superset-frontend/.eslintrc.js

```diff
@@ -33,7 +33,44 @@ module.exports = {
     browser: true,
   },
   settings: {
-    'import/resolver': 'webpack',
+    'import/resolver': {
+      webpack: {},
+      node: {
+        extensions: ['.js', '.jsx', '.ts', '.tsx', '.json'],
+      },
+    },
+    // Allow only core/src and core/test, avoid import modules from lib
+    'import/internal-regex': /^@superset-ui\/core\/(src|test)\/.*/,
+    'import/core-modules': [
```

Comment:
Would it make sense to separate `core-modules` (core, chart-controls) and `plugins` (everything else) here?

Reply:
This is for
- **Allow** `import { variable } from '@superset-ui/core/src/some/other/module'` and `import { variable } from '@superset-ui/core/test/some/other/module'`
- **Disallow** `import { variable } from '@superset-ui/core/lib/some/other/module'` or `import { variable } from '@superset-ui/chart-controls/lib/some/other/module'`

We should use `@superset-ui/core` instead of `@superset-ui/core/src/some/other/module` or`@superset-ui/core/test/some/other/module` after monorepo project.